frontend/src/components/Folders/Folder.jsx

```diff
@@ -1,12 +1,121 @@
+import { useEffect, useState } from 'react'
 import './Folder.css'
+import { doc, updateDoc, arrayUnion, getDoc, setDoc, collection, addDoc } from 'firebase/firestore';
+import { db } from '../../../../backend/firebase/firebase.config';
+import { useAuth } from '../auth/auth.jsx';
 
+const Folder = ({ folderName }) => {
+    const { currentUser } = useAuth();
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [transcripts, setTranscripts] = useState([]);
+    const [isTranscriptOpen, setIsTranscriptOpen] = useState(false);
+    const [currentTranscriptName, setCurrentTranscriptName] = useState("");
+    const [transcriptData, setTranscriptData] = useState([]);
 
 
-const Folder = ({folderName}) => {
+    const getTranscripts = async () => {
+        const docRef = doc(db, "users", currentUser.uid)
+        const userDoc = await getDoc(docRef);
+        if (!userDoc.exists()) {
+            console.error("User document could not be found.");
+        }
+        const userData = userDoc.data();
+        const userTranscripts = userData.transcripts;
+        const transcriptNames = userTranscripts[folderName];
+        setTranscripts(transcriptNames || []);
+    }
+
+    const handleOpenModal = async () => {
+        getTranscripts();
+        setIsModalOpen(true);
+
+    }
+    useEffect(() => {
+        setTranscripts([]);
+    }, [folderName])
+
+
+    const viewTranscript = async (transcriptName) => {
+        setIsModalOpen(false);
+        setIsTranscriptOpen(true);
+        const userRef = doc(db, "users", currentUser.uid, folderName, transcriptName);
+        const currentTranscript = await getDoc(userRef)
+        if (!currentTranscript.exists()) {
+            console.error("This transcript could not be found.");
+        }
+        const docData =   currentTranscript.data()
+        setCurrentTranscriptName(transcriptName)
+        setTranscriptData(docData.transcript)
+        // transcriptData.map((message, index) => {
+        //     console.log("Message:", message.content);
+        // })
```

Comment:
Remove if not planning to use. Add descriptive comment explaining why you might need this otherwise.

Reply:
Was removed.